torchdynamo/eval_frame.py

```diff
@@ -132,7 +134,29 @@ def __init__(self):
 def catch_errors_wrapper(callback):
     @functools.wraps(callback)
     def catch_errors(frame, cache_size):
+        def _inside_ddp_callstack(frame):
+            """
+            Totally braindead way to determine we are inside a DDP forward.
+
+            TODO replace this with some kind of push/pop context that keeps track
+            efficiently.
+            """
+            up = frame.f_back
+            while up:
+                if up.f_code.co_name == "_run_ddp_forward":
+                    return True
+                up = up.f_back
+            return False
+
         try:
+            if _inside_ddp_callstack(frame) and frame.f_code.co_name == "forward":
+                print(f"compile for ddp: {frame.f_code.co_name} {frame.f_code.co_filename}")
+                with compile_lock:
+                    # TODO(whc) find a way to get these parameters from the DDP module
+                    # and configure the backend compiler and convert_frame correctly
+                    ddp_optimizer = DDPOptimizer(bucket_cap_mb = 25, parameters_to_ignore = [], backend_compile_fn = BACKENDS["aot_autograd"])
```

Comment:
@jansel any ideas how to achieve these few lines worth of stuff in a more sane way?

Reply:
- how to detect 'inside ddp'
- how to access attributes of the DDP module object (such as its 'bucket_cap_mb' attr)
- how to get the user-provided backend compile_fn rather than hardcoding aot_autograd here
- whether the hijacked_callback approach is even correct?